### PR TITLE
feat: prometheus monitor_kubernetes_pods add setting to change namespace label name of the pod to avoid conflicts

### LINKED ISSUE #11383: Prometheus monitor_kubernetes_pods overwrites the namespace tag for a metric with the namespace of the pod being scraped
### Relevant telegraf.conf

```toml
[[inputs.prometheus]]
  interval = "1m"
  monitor_kubernetes_pods = true
  pod_scrape_scope = "node"
  kubernetes_label_selector = ""
  kubernetes_field_selector = ""
  fieldpass = ["kube_pod_info", "kube_pod_container_info"]
  fielddrop = []
  metric_version = 2
  url_tag = "scrapeUrl"
  # kube_config = "/path/to/kubernetes.config"
  bearer_token = "/var/run/secrets/kubernetes.io/serviceaccount/token"
  response_timeout = "15s"
  tls_ca = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
  insecure_skip_verify = true
```

### Logs from Telegraf

```text
N/A
```

### System info

Telegraf 1.22.2, Ubuntu 18.04, Kubernetes v1.21.7

### Docker

_No response_

### Steps to reproduce

1. Set `monitor_kubernetes_pods=true` and use pod annotations to scrape a Prometheus endpoint from a pod that has metrics with `namespace` as a dimension name.
2. Deploying `kube-state-metrics` is an easy way to test by adding the annotations to the pod:
        prometheus.io/port: '8080'
        prometheus.io/scrape: 'true'
Note: For `kube-state-metrics`, to get around this, the `kubernetes_services` setting can be used. However, this is used as an easy example and this will not always work. For example Kubevirt (https://github.com/kubevirt/kubevirt) does not have an associated service and all pods in the daemonset need to be scraped

### Expected behavior

The dimension value for `namespace` shouldn't be overwritten by the namespace of the pod that the metrics are being scraped from. For example, if a `kube-state-metrics` pod in the `kube-system` namespace is being scraped, the metric for the pod named `example-pod` in `example-namespace` should be: `kube_pod_info{pod="example-pod", namespace="example-namespace"}`

### Actual behavior

The dimension value for `namespace` is the namespace of the pod that the metrics are being scraped from instead of the actual value of the metric for the namespace dimension. For example, if a `kube-state-metrics` pod in the `kube-system` namespace is being scraped, the metric for the pod named `example-pod` in `example-namespace` is: `kube_pod_info{pod="example-pod", namespace="kube-system"}`

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by gracewehner**
# Required for all PRs

- [X] Updated associated README.md.
- [X] Wrote appropriate unit tests.
- [X] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves #11383

When using `monitor_kubernetes_pods` for Prometheus input plugin, currently the label `namespace` is added for the pod that is scraped. This however overwrites the `namespace` label if a metric that is scraped on that pod has that label. This PR adds a new setting that allows you to change the namespace label name for the pod so that this conflict does not occur.

### CODE REVIEW

File: plugins/inputs/prometheus/README.md [hunk 1]
```diff
@@ -53,6 +53,9 @@ in Prometheus format.
   ## Restricts Kubernetes monitoring to a single namespace
   ##   ex: monitor_kubernetes_pods_namespace = "default"
   # monitor_kubernetes_pods_namespace = ""
+  ## The name of the label for the pod that is being scraped.
+  ## Default is 'namespace' but this can conflict with metrics that have the label 'namespace'
+  # pod_namespace_label_name = "pod_namespace"
```
Comment (srebhan) [inline plugins/inputs/prometheus/README.md:58]: If the default is `"namespace"`, please use
```suggestion
  # pod_namespace_label_name = "namespace"
```

File: plugins/inputs/prometheus/sample.conf [hunk 2]
```diff
@@ -45,6 +45,9 @@
   ## Restricts Kubernetes monitoring to a single namespace
   ##   ex: monitor_kubernetes_pods_namespace = "default"
   # monitor_kubernetes_pods_namespace = ""
+  ## The name of the label for the pod that is being scraped.
+  ## Default is 'namespace' but this can conflict with metrics that have the label 'namespace'
+  # pod_namespace_label_name = "pod_namespace"
```
Comment (srebhan) [inline plugins/inputs/prometheus/sample.conf:50]:
```suggestion
  # pod_namespace_label_name = "namespace"
```

**Review by srebhan — COMMENTED** [review]
Thanks @gracewehner for the contribution! I've just one comment regarding the default as this is the way we usually handle defaults...

**Review by srebhan — APPROVED** [review]
Looks good to me. Thanks for your contribution @gracewehner!

**Review by powersj — APPROVED** [review]
Thank you! I know others will find this helpful

**Pull request merged**